plugin.go

```diff
@@ -0,0 +1,137 @@
+// Package plasmactlupdate implements an update launchr plugin
+package plasmactlupdate
+
+import (
+	"fmt"
+	"github.com/launchrctl/keyring"
+	"github.com/launchrctl/launchr"
+	"github.com/spf13/cobra"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func init() {
+	launchr.RegisterPlugin(&Plugin{})
+}
+
+// Plugin is launchr plugin providing update action.
+type Plugin struct{}
+
+// PluginInfo implements launchr.Plugin interface.
+func (p *Plugin) PluginInfo() launchr.PluginInfo {
+	return launchr.PluginInfo{}
+}
+
+// OnAppInit implements launchr.Plugin interface.
+func (p *Plugin) OnAppInit(_ launchr.App) error {
+	return nil
+}
+
+// CobraAddCommands implements launchr.CobraPlugin interface to provide bump functionality.
+func (p *Plugin) CobraAddCommands(rootCmd *cobra.Command) error {
+	var creds keyring.CredentialsItem
+
+	var updCmd = &cobra.Command{
+		Use:   "update",
+		Short: "Command to fetch and install latest version of plasmactl",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Don't show usage help on a runtime error.
+			cmd.SilenceUsage = true
+			u, err := CreateUpdate(creds)
+			if err != nil {
+				return err
+			}
+
+			return runUpdate(u)
+		},
+	}
+
+	// Credentials flags
+	creds.URL = BaseUrl
+	updCmd.Flags().StringVarP(&creds.Username, "username", "u", "", "Username")
+	updCmd.Flags().StringVarP(&creds.Password, "password", "p", "", "Password")
+	rootCmd.AddCommand(updCmd)
+
+	return nil
+}
+
+// runUpdate command entrypoint.
+func runUpdate(u *Update) error {
+	// Wrapper to conclude errors.
+	if err := runCommands(u); err != nil {
+		u.exitWithError()
+		return err
+	}
+
+	return nil
+}
+
+// runCommands run commands one by one.
+func runCommands(u *Update) error {
+	currOS, arch, err := u.initVars()
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("Starting plasmactl installation...")
```

Comment:
Would be great for this string to be displayed before credentials are asked

![image](https://github.com/skilld-labs/plasmactl-update/assets/20738264/c4f4ceab-1eb3-4ce9-8352-9cd9b2f60b54)